Updated Frontend on a bunch of the pages

RafiPibe opened the pull request:
should be functioning (p.s dont go rough pls)

BenGhahramani reviewed (changes requested):
Duplicate <body> headers across create project, teacher dahsboard, view project and login html. Should probably keep a single html head body and one header per page and remove the legacy dark theme blocks.

UI calls PATCH which doesnt exist in server. Probably want to use existing PUT / id or add required end points.

There also seems to be both in memory and JSON file implemenbtations, we should use one canonical set of routes. 

add .ds_store to git ignore we shouldve done this at the start lowkey 

I could also just be wrong about some of this also

RafiPibe commented:
## Major changes

### UI Changes and implementation from Figma

**Edited**
- `login.html`
- `create-project.html`
- `teacher-dashboard.html`
- `view-project.html`

**Added**
- `global.css`
- `header.html`

## Details

- Added updates towards the .gitignore for unwanted files
- Added global and linked properly towards corresponding files
- Added global header and linked with js
- changed PATCH method to the existing PUT method
- file implementation should already be in JSON

## Justification
- Tried implementing global head html, though it cannot detect the tailwind script, thus having to rely on individual head in the meantime
- Will try to implement the other pages first before editing the frontend of `ar-demo.html` and `blockly-demo.html`

BenGhahramani reviewed (approved):
lgtm
